tsconfig.json

```diff
@@ -1,5 +1,47 @@
 {
   "compilerOptions": {
+    /* Visit https://aka.ms/tsconfig to read more about this file */
+    /* Language and Environment */
+    "target": "ES2021", /* Set the JavaScript language version for emitted JavaScript and include compatible library declarations. */
+    "lib": [
+      "DOM"
+    ], /* Specify a set of bundled library declaration files that describe the target runtime environment. */
+    /* Modules */
+    "module": "ESNext", /* Specify what module code is generated. */
+    "moduleResolution": "node", /* Specify how TypeScript looks up a file from a given module specifier. */
```

Comment:
For the `module`, `nodenext` seems to be capable of handling ESM and CJS. 
Also from the doc on `ESNext`:  `Do not use for Node.js. Use node16 or nodenext with "type": "module" in package.json to emit ES modules for Node.js.`.
Even if we don't emit files,  we should switch module to `nodenext` instead of `esnext`, WDYT?

Reply:
ah yes good idea, I just need to check is `ESNext` is supported by `typescritpt@4.3.2` (if not, we'll have to use `node` which should be equal to `node16` :thinking:)